rpc/handlers_test.go

```diff
@@ -99,3 +99,21 @@ func TestThrottledVMError(t *testing.T) {
 		assert.Equal(t, throttledErr, rpcErr.Data)
 	})
 }
+
+func TestJunoGetBlockFromRoot(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	t.Cleanup(mockCtrl.Finish)
+	mockReader := mocks.NewMockReader(mockCtrl)
+	mockReader.EXPECT().Network().Return(&utils.Mainnet).AnyTimes()
+	log := utils.NewNopZapLogger()
+	mockState := mocks.NewMockStateHistoryReader(mockCtrl)
+
+	handler := rpc.New(mockReader, nil, nil, "", utils.Ptr(utils.Mainnet), log)
+
+	t.Run("Key DNE", func(t *testing.T) {
```

Comment:
This is panicking because you are calling 
// NodeFromRoot implements core.StateHistoryReader.
func (m *MockStateHistoryReader) NodeFromRoot() (*trie.Trie, func() error, error) {
	panic("unimplemented")
}